### ISSUE TITLE
Textless NLP / dGSLM: the pre-trained  SpeechDLM model with poor performance

### ISSUE THREAD

**Issue by mkd-mkd** [facebookresearch/fairseq#5041]
### Before asking:
1. search the issues.
2. search the docs.

#### What is your question?
The pre-trained SpeechDLM model gets poor performance on the Fisher valid dataset. 

Moreover, when loading the checkpoint, there is no criterion named 'duration_cross_multichannel_cross_entropy'. The `state['cfg']['criterion']` of the checkpoint is `{'_name': 'duration_cross_multichannel_cross_entropy', 'sentence_avg': False, 'main_and_cross_weights': '1,0', 'include_edge_loss': True, 'edge_loss_weight': 1.0, 'general_loss_weight': 0.0, 'normalize_edge_loss': True, 'duration_loss_weight': 1.0, 'shift_duration_target': True, 'round_duration_prediction': False, 'normalize_duration_loss': False}`.

We replaced the `state['cfg']['criterion']` with` {'_name': 'speech_dlm_criterion', 'sentence_avg': False, 'main_and_cross_weights': '1,0', 'general_unit_loss_weight': 0.0, 'edge_unit_loss_weight': 1.0, 'duration_loss_weight': 1.0}`.

Then the checkpoint can be loaded sucessfully but the loss on the valid dataset is abnormal. The loss is shown as follows. Is there any wrong with the checkpoint?

#### Code
We run the command
```
CUDA_VISIBLE_DEVICES=0,1 fairseq-validate {BIN_DATA_DIR} \
    --task speech_dlm_task \
    --path {CHECKPOINT_DIR} \
    --max-tokens 6144
```
and the result shows "2023-03-23 16:23:22 | INFO | valid | valid on 'valid' subset | [unitA-unitA]edge_token_loss 9.351 | [unitA-unitA]edge_token_pred_acc 8.689 | [unitA-unitA]edge_duration_loss 1.324 | [unitA-unitA]edge_duration_pred_acc 52.054 | [unitB-unitB]edge_token_loss 9.418 | [unitB-unitB]edge_token_pred_acc 7.937 | [unitB-unitB]edge_duration_loss 1.39 | [unitB-unitB]edge_duration_pred_acc 52.463 | edge_token_loss 22.028 | edge_token_pred_acc 8.317 | edge_duration_loss 3.441 | edge_duration_pred_acc 52.256 | loss 11.459 | [unitA-unitA]edge_token_ppl 653.15 | [unitB-unitB]edge_token_ppl 684.18 | edge_token_ppl 4.27656e+06 | wps 0 | wpb 6.40306e+06 | bsz 522".

#### What have you tried?
read the code and paper, and check the dataset.
#### What's your environment?

 - fairseq Version (e.g., 1.0 or main):`0.12.2`
 - PyTorch Version (e.g., 1.0):`1.12`
 - OS (e.g., Linux):`Ubuntu 18.04.4 LTS`
 - How you installed fairseq (`pip`, source): `source`
 - Build command you used (if compiling from source): `pip3 install --editable ./`
 - Python version: `3.9.13`
 - CUDA/cuDNN version:`CUDA10.2`
 - GPU models and configuration:`NVIDIA RTX 2080Ti`
 - Any other relevant information:-

**Comment by mkd-mkd** [comment]
I'm sorry, it is my fault

**Comment by tuanh208** [comment]
Thanks for pointing out the criterion name mismatch. I have updated the criterion in the model checkpoint.
